pulsar-connect/core/src/main/java/org/apache/pulsar/connect/core/Sink.java

```diff
@@ -42,8 +42,8 @@
     /**
      * Attempt to publish a type safe collection of messages
      *
-     * @param message Object to publish to the sink
+     * @param outputValue output value
      * @return Completable future fo async publish request
      */
-    CompletableFuture<Void> write(final Message<T> message);
+    CompletableFuture<Void> write(T outputValue);
```

Comment:
Shouldn't this be wrapped around the a Record object?

Reply:
Record should be something from source only. T is the value passed on to the sink, sink can convert it to the format that it can use to write.

Reply:
The Source will always return Record<T> from read though.  Now the generic T in the sink can represent Record<T>. Isn't that inconsistent from the Source interface.   Users might find that confusing. And in what circumstance is the outputValue in the sink write method not a Record object?

Reply:
@jerrypeng I think in your pull request, this method will become `write(Record<?> inputRecord, T outputValue)`, no? In this case, T is the result after processing `inputRecord`. at this moment, we have no idea about which partition, what sequence id we want to write for the result, no?